### PR TITLE
PAYARA-3885 @RolesPermitted Dynamic role feature testcases

### PR DESCRIPTION
**Pull request by jGauravGupta**

### CODE REVIEW

File: samples/dynamic-roles/src/test/java/fish/payara/samples/dynamic/roles/PersonControllerClientHelper.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            builder.register((ClientRequestFilter) context -> {
+                context.getHeaders().add("username", DEFAULT_USER);
+                context.getHeaders().add("password", DEFAULT_PASSWORD);
+            });
+            PersonControllerClient client = builder
+                    .baseUrl(new URL(deploymentUrl.toURI().toString() + "resources/"))
+                    .build(PersonControllerClient.class);
+            return client;
+        } catch (URISyntaxException | MalformedURLException ex) {
+            throw new RuntimeException(ex);
```
Comment (dmatej) [inline samples/dynamic-roles/src/test/java/fish/payara/samples/dynamic/roles/PersonControllerClientHelper.java]: I don't know how is SonarQube rules set here, but usually is better to use any of "semantic" exceptions, so in this case IllegalStateException would be better than generic RuntimeException.
See https://rules.sonarsource.com/java/RSPEC-112

File: samples/dynamic-roles/src/main/java/fish/payara/samples/dynamic/roles/ejb/PersonService.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.samples.dynamic.roles.ejb;
+
+import fish.payara.samples.dynamic.roles.common.Person;
+import fish.payara.cdi.auth.roles.RolesPermitted;
+import static fish.payara.samples.dynamic.roles.common.AuthoritiesConstants.ADMIN;
```
[the remark this answers is not in the document]
Comment (jGauravGupta) [reply samples/dynamic-roles/src/main/java/fish/payara/samples/dynamic/roles/ejb/PersonService.java]: Yes, NetBeans IDE supports automatic organization of imports in alphabetical order but not with specific preference to static type.

**Comment by MarkWareham** [comment]
Requested Matt for review as he changed config of poms recently

File: pom.xml [hunk 3]
```diff
@@ -164,6 +164,12 @@
             <version>8.0</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
```
Comment (MattGill98) [inline pom.xml]: This needs a merge from master, the parent should have no dependencies now

File: samples/dynamic-roles/pom.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    <packaging>war</packaging>
+    <name>Payara Samples - Payara - Dynamic Roles</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.eclipse.microprofile</groupId>
+            <artifactId>microprofile</artifactId>
+            <type>pom</type>
+            <scope>provided</scope>
+            <version>2.2</version>
```
Comment (MattGill98) [inline samples/dynamic-roles/pom.xml]: I've been meaning to do this so call me lazy, but would you be able to put this in the dependency management and remove the opentracing dependencies?

File: samples/dynamic-roles/pom.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            <groupId>io.smallrye</groupId>
+            <artifactId>smallrye-config</artifactId>
+            <scope>test</scope>
+            <version>1.3.7</version>
+        </dependency>
+        <dependency>
+            <groupId>fish.payara.arquillian</groupId>
+            <artifactId>payara-client-ee8</artifactId>
+            <scope>test</scope>
+            <version>1.1</version>
```
Comment (MattGill98) [inline samples/dynamic-roles/pom.xml]: I believe this version is already specified in the parent pom

**Pull request merged**